src/Microsoft.CodeAnalysis.Remote.Razor/RazorLanguageService.cs

```diff
@@ -19,25 +21,69 @@ internal class RazorLanguageService : ServiceHubServiceBase
         public RazorLanguageService(Stream stream, IServiceProvider serviceProvider)
             : base(serviceProvider, stream)
         {
-            Rpc.JsonSerializer.Converters.Add(new RazorDiagnosticJsonConverter());
+            Rpc.JsonSerializer.Converters.Add(TagHelperDescriptorJsonConverter.Instance);
+            Rpc.JsonSerializer.Converters.Add(RazorDiagnosticJsonConverter.Instance);
+            Rpc.JsonSerializer.Converters.Add(RazorExtensionJsonConverter.Instance);
+            Rpc.JsonSerializer.Converters.Add(RazorConfigurationJsonConverter.Instance);
+            Rpc.JsonSerializer.Converters.Add(ProjectSnapshotJsonConverter.Instance);
 
             // Due to this issue - https://github.com/dotnet/roslyn/issues/16900#issuecomment-277378950
             // We need to manually start the RPC connection. Otherwise we'd be opting ourselves into 
             // race condition prone call paths.
             Rpc.StartListening();
+
+            Debugger.Launch();
         }
 
-        public async Task<TagHelperResolutionResult> GetTagHelpersAsync(Guid projectIdBytes, string projectDebugName, CancellationToken cancellationToken = default(CancellationToken))
+        public async Task<TagHelperResolutionResult> GetTagHelpersAsync(
+            Guid projectIdBytes, 
+            string projectDebugName, 
+            string factoryTypeName, 
+            ProjectSnapshot snapshot, 
+            CancellationToken cancellationToken = default(CancellationToken))
         {
             var projectId = ProjectId.CreateFromSerialized(projectIdBytes, projectDebugName);
 
             var solution = await GetSolutionAsync(cancellationToken).ConfigureAwait(false);
             var project = solution.GetProject(projectId);
 
-            var resolver = new DefaultTagHelperResolver(designTime: true);
-            var result = await resolver.GetTagHelpersAsync(project, cancellationToken).ConfigureAwait(false);
+            ((SerializedProjectSnapshot)snapshot).InitializeWorkspaceProject(project);
+
+            RazorTemplateEngine templateEngine = null;
+            if (factoryTypeName != null)
+            {
+                var factoryType = Type.GetType(factoryTypeName, throwOnError: true);
+                var factory = (ICustomTemplateEngineFactory)Activator.CreateInstance(factoryType);
+
+                templateEngine = factory.Create(snapshot.Configuration, EmptyProject.Instance, (b) => 
```

Comment:
Could always get the project path from the project snapshot instead of needing an empty project instance.

Reply:
We need a project to create a template engine.

Reply:
Ya you can do a `RazorProject.Create(...)` from the project snapshots file path.

Reply:
Ah I see what you mean! Yeah I did the empty thing on purpose because I don't want any code running here to look at the files.

Reply:
> I don't want any code running here to look at the files.

O rly? Why's that?

Reply:
Because we don't have a clear need for it.